### DISCUSSION TITLE
rsa: encrypting (private key) ->decrypting (public key) doesn't work anymore?

### DISCUSSION THREAD

**Discussion by MRX418** [openssl/openssl#23733] [Q&A] [answered]
I am in a context where I really need to use a PRIVATE key to ENCRYPT a symmetric key and use a PUBLIC key to DECRYPT it. The code works fine with `rsa_private_encrypt` and `rsa_public_decrypt`, but these are deprecated functions. I'm trying to find a way to achieve the exact same behavior with non-deprecated functions. Yes, I am aware it's not secure to practice this, but my context doesn't allow me to use decryption with the private key and encryption with the public key. I am also aware that `EVP_sign` and` EVP_verify` are the newest closest functions to what I'm trying to do, but verification doesn't return decrypted data. `EVP_encrypt` and `EVP_decrypt` only work when I encrypt with the public key and decrypt with the private key and not vice versa, so I'm kind of stuck here. HELP please!

**Comment by t8m** [comment, marked as answer]
There is an EVP_PKEY_verify_recover() function that should provide you the access to the decrypted data.
